File: solutions/typescript/darts/1/darts.ts
export function score(x: number, y: number): number {
  let points = 0;
  
  const radius = Math.sqrt(Math.pow(x, 2) + Math.pow(y, 2));
  if(radius <= 1){
    points = 10;
  }
  else if(radius <= 5){
    points = 5;
  }
  else if(radius <= 10){
    points = 1;
  }

  return points;
}

File: solutions/java/darts/1/src/main/java/Darts.java
class Darts {
    int score(double xOfDart, double yOfDart) {
        int score = 0;
        double radius = this.calculateRadius(xOfDart, yOfDart);
        
        if(radius <= 1){
            score = 10;
        }
        else if(radius <= 5){
            score = 5;
        }
        else if(radius <= 10){
            score = 1;
        }
        
        return score;
    }

    private double calculateRadius (double x, double y){
        return Math.sqrt(x*x + y*y);
    }
}